### DISCUSSION TITLE
Continuously varying drift rates based on the stimulus

### DISCUSSION THREAD

**Discussion by ncpuneeth** [lnccbrown/HSSM#307] [Q&A]
Dear all,

I am new to using the HSSM package. I went through the documentation, but I am not able to figure out if and how we can fit a HDDM model with a continuously varying drift rate that depends on the stimulus (which is not static). In the examples shown, there is one response (decision) for one stimulus parameter, but what I am looking for is to model a task where the decision is made after sequentially sampling many stimuli that continuously vary the drift rate. Any help on how the parameters should be set up for fitting such a model is highly appreciated. Thank you!

**Comment by frankmj** [comment]
Hi there,

when you mention the stimulus is not static do you mean within a
trial, where the stimulus presented at any moment in time could shift the
drift one direction or the other? Or do you mean that there is noisy
evidence but which could be fit with a single drift rate per stimulus
condition? If the latter, then you can fit this with HSSM using a
regression, whereby v is fit as a function of stimulus condition
(either categorically or continuously). But if the former, then that
requires a different likelihood. If the stimuli are presented at fixed
intervals within a trial then it would be feasible to train a LAN to learn
that likelihood function in which v switches directions at those intervals.
But that would require the LAN factory. If the stimuli are presented at
arbitrary times across trials, then this is more complex (and is similar to
the aDDM - where drift varies as a function of visual fixations that can
occur randomly within a trial). There are approximations to this but we are
currently working on a method to obtain an efficient likelihood for such
scenarios. Hopefully to be included in a future release of hSSM.

Michael

Michael J Frank, PhD | Edgar L. Marston Professor
Director, Carney Center for Computational Brain Science
<https://www.brown.edu/carney/ccbs>
Laboratory of Neural Computation and Cognition <https://www.lnccbrown.com/>

Brown University
website <http://ski.clps.brown.edu>
